File: AtCoder/ABC/ABC229/C.py
n, w = map(int,input().split())
a = []
for i in range(n):
    x, y = map(int,input().split())
    a.append((x, y))
a.sort(reverse=True, key=lambda x: x[0])

now = 0
ans = 0
for i in range(n):
    if now >= w:
        break
    if now + a[i][1] <= w:
        now += a[i][1]
        ans += a[i][0] * a[i][1]
    else:
        ok = w - now
        ans += a[i][0] * ok
        now = w

print(ans)

File: AtCoder/ABC/ABC229/C.js
function main(input) {
    inputs = input.split('\n');
    const n = parseInt(inputs[0].split(' ')[0]);
    let w = BigInt(inputs[0].split(' ')[1]);

    const ab = [];
    for (let i = 0; i < n; i ++) {
        const a = BigInt(inputs[i + 1].split(' ')[0]);
        const b = BigInt(inputs[i + 1].split(' ')[1]);

        ab.push([a, b]);

    }

    ab.sort((a, b) => Number(b[0] - a[0]));

    let ans = 0n;
    for (let i = 0; i < n; i ++) {
        if (w - ab[i][1] >= 0) {
            ans += ab[i][0] * ab[i][1];
            w -= ab[i][1];
        } else {
            ans += ab[i][0] * w;
            break; 
        }
    }

    console.log(ans.toString());

}

main(require('fs').readFileSync('/dev/stdin', 'utf8')); 